client/lib/signup/step-actions.js

```diff
@@ -28,6 +28,19 @@ import {
 import { getSiteTitle } from 'state/signup/steps/site-title/selectors';
 import { getSurveyVertical, getSurveySiteType } from 'state/signup/steps/survey/selectors';
 
+function createCart( callback, dependencies, data ) {
+	const { designType } = dependencies;
+	const { domainItem, themeItem } = data;
+
+	if ( designType === 'domain' ) {
+		// TODO: Find a way to create cart without a site
+		// SignupCart.addToCart depends on the site's slug
+		callback( undefined, [ null, null, domainItem, themeItem ] );
```

Comment:
It's tricky because we now take advantage of site slug and redirect users to `/checkout/:siteSlug` url when an item is added to the cart. It looks like we need to build new page from scratch. It would work only with a single domain and wouldn't need to have connected site.